### ISSUE TITLE
Heads-up: using the Job source code in Redux Toolkit

### ISSUE THREAD

**Issue by markerikson** [ethossoftworks/job-ts#1]
Hi, I'm the primary Redux maintainer and creator of Redux Toolkit. I've been working on designing a new side effects middleware to include in RTK (per https://github.com/reduxjs/redux-toolkit/discussions/1648 ), and one of the capabilities I've wanted to include is some kind of a hierarchical job system with cancelation.  After digging through Github and NPM, I ran across your package, and over the last couple days I've been playing with integrating it into the WIP middleware:

https://github.com/reduxjs/redux-toolkit/pull/1792

So far, it's looking _very_ promising! Still have a lot of work to do, but I think this is very much the right set of capabilities I was looking for: lightweight, async/await-based, and with the ability to spawn child tasks and cancel them.

I've copied over the source for both `jobs.ts` and `outcome.ts` into our repo, because A) I'm making some changes to the implementation (modified a couple functions, and may end up trying to rewrite it as a closure instead of class), and B) I'm trying to minimize additional dependency packages.

Both your packages are MIT licensed and I've already referenced the origin repos in the source.  But, as a courtesy, I wanted to express my thanks for writing these libs and publishing them, and give you a heads-up that I'm making use of your code.

Thank you!

**Comment by ryanmitchener** [comment]
@markerikson I'm unfortunately just now seeing this. I see that you are not using my code now but nevertheless, thank you for the kind words and the heads up!

**Comment by markerikson** [comment]
Yeah, we ended up switching over to an implementation based on `AbortController`.  

If you're curious, the whole backstory of how we developed this feature over a couple years is here:

- https://blog.isquaredsoftware.com/2022/03/designing-rtk-listener-middleware/

And the current implementation lives here:

- https://github.com/reduxjs/redux-toolkit/blob/v1.9.5/packages/toolkit/src/listenerMiddleware/index.ts

Thanks for the inspiration!  Really came in helpful.
